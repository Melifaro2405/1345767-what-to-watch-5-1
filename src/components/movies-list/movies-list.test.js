import React from 'react';
import renderer from 'react-test-renderer';
import MoviesList from './movies-list';

jest.mock(`../small-movie-card/small-movie-card`, () => `SmallMovieCard`);

const films = [
  {
    id: 1,
    preview: {
      src: `img/the-grand-budapest-hotel-poster.jpg`,
      title: `The Grand Budapest Hotel`
    },
    moreInfo: {
      backgroundSrc: `img/the-grand-budapest-hotel-bg.jpg`,
      backgroundColor: `#ffffff`,
      posterSrc: `img/the-grand-budapest-hotel-poster.jpg`,
      genre: `Comedy`,
      releaseDate: 2014,
      playPreviewSrc: `https://some-link`,
      playVideoSrc: `https://some-link`,
      isAddToMyList: false
    },
    overview: {
      description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
        presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy,
        becomes Gustave's friend and protege.`,
      rating: 8.9,
      ratingCount: 240,
      director: `Wes Andreson`,
      actorsList: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`]
    },
    details: {
      runtime: 99
    }
  },
  {
    id: 2,
    preview: {
      src: `img/the-grand-budapest-hotel-poster.jpg`,
      title: `The Grand Budapest Hotel`
    },
    moreInfo: {
      backgroundSrc: `img/the-grand-budapest-hotel-bg.jpg`,
      backgroundColor: `#ffffff`,
      posterSrc: `img/the-grand-budapest-hotel-poster.jpg`,
      genre: `Comedy`,
      releaseDate: 2014,
      playPreviewSrc: `https://some-link`,
      playVideoSrc: `https://some-link`,
      isAddToMyList: false
    },
    overview: {
      description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
        presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy,
        becomes Gustave's friend and protege.`,
      rating: 8.9,
      ratingCount: 240,
      director: `Wes Andreson`,
      actorsList: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`]
    },
    details: {
      runtime: 99
    }
  },
  {
    id: 3,
    preview: {
      src: `img/the-grand-budapest-hotel-poster.jpg`,
      title: `The Grand Budapest Hotel`
    },
    moreInfo: {
      backgroundSrc: `img/the-grand-budapest-hotel-bg.jpg`,
      backgroundColor: `#ffffff`,
      posterSrc: `img/the-grand-budapest-hotel-poster.jpg`,
      genre: `Comedy`,
      releaseDate: 2014,
      playPreviewSrc: `https://some-link`,
      playVideoSrc: `https://some-link`,
      isAddToMyList: false
    },
    overview: {
      description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
        presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy,
        becomes Gustave's friend and protege.`,
      rating: 8.9,
      ratingCount: 240,
      director: `Wes Andreson`,
      actorsList: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`]
    },
    details: {
      runtime: 99
    }
  }
];

it(`Should MoviesList render correctly`, () => {
  const tree = renderer
    .create(
        <MoviesList
          films={films}
        />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
