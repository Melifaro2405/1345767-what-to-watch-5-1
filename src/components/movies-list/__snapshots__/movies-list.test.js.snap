// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MoviesList render correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <SmallMovieCard
    film={
      Object {
        "details": Object {
          "runtime": 99,
        },
        "id": 1,
        "moreInfo": Object {
          "backgroundColor": "#ffffff",
          "backgroundSrc": "img/the-grand-budapest-hotel-bg.jpg",
          "genre": "Comedy",
          "isAddToMyList": false,
          "playPreviewSrc": "https://some-link",
          "playVideoSrc": "https://some-link",
          "posterSrc": "img/the-grand-budapest-hotel-poster.jpg",
          "releaseDate": 2014,
        },
        "overview": Object {
          "actorsList": Array [
            "Bill Murray",
            "Edward Norton",
            "Jude Law",
            "Willem Dafoe",
            "Saoirse Ronan",
          ],
          "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
        presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy,
        becomes Gustave's friend and protege.",
          "director": "Wes Andreson",
          "rating": 8.9,
          "ratingCount": 240,
        },
        "preview": Object {
          "src": "img/the-grand-budapest-hotel-poster.jpg",
          "title": "The Grand Budapest Hotel",
        },
      }
    }
    isPlayingVideo={false}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  />
  <SmallMovieCard
    film={
      Object {
        "details": Object {
          "runtime": 99,
        },
        "id": 2,
        "moreInfo": Object {
          "backgroundColor": "#ffffff",
          "backgroundSrc": "img/the-grand-budapest-hotel-bg.jpg",
          "genre": "Comedy",
          "isAddToMyList": false,
          "playPreviewSrc": "https://some-link",
          "playVideoSrc": "https://some-link",
          "posterSrc": "img/the-grand-budapest-hotel-poster.jpg",
          "releaseDate": 2014,
        },
        "overview": Object {
          "actorsList": Array [
            "Bill Murray",
            "Edward Norton",
            "Jude Law",
            "Willem Dafoe",
            "Saoirse Ronan",
          ],
          "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
        presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy,
        becomes Gustave's friend and protege.",
          "director": "Wes Andreson",
          "rating": 8.9,
          "ratingCount": 240,
        },
        "preview": Object {
          "src": "img/the-grand-budapest-hotel-poster.jpg",
          "title": "The Grand Budapest Hotel",
        },
      }
    }
    isPlayingVideo={false}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  />
  <SmallMovieCard
    film={
      Object {
        "details": Object {
          "runtime": 99,
        },
        "id": 3,
        "moreInfo": Object {
          "backgroundColor": "#ffffff",
          "backgroundSrc": "img/the-grand-budapest-hotel-bg.jpg",
          "genre": "Comedy",
          "isAddToMyList": false,
          "playPreviewSrc": "https://some-link",
          "playVideoSrc": "https://some-link",
          "posterSrc": "img/the-grand-budapest-hotel-poster.jpg",
          "releaseDate": 2014,
        },
        "overview": Object {
          "actorsList": Array [
            "Bill Murray",
            "Edward Norton",
            "Jude Law",
            "Willem Dafoe",
            "Saoirse Ronan",
          ],
          "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
        presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy,
        becomes Gustave's friend and protege.",
          "director": "Wes Andreson",
          "rating": 8.9,
          "ratingCount": 240,
        },
        "preview": Object {
          "src": "img/the-grand-budapest-hotel-poster.jpg",
          "title": "The Grand Budapest Hotel",
        },
      }
    }
    isPlayingVideo={false}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  />
</div>
`;
